src/lifecycle/properties-init.js

```diff
@@ -46,6 +46,11 @@ function createNativePropertyDefinition (name, opts) {
     const attributeName = getLinkedAttribute(name, opts.attribute);
     let initialValue = elem[name];
 
+    const observedAttributes = elem.constructor.observedAttributes;
```

Comment:
It'd be ideal to do this up front when the element is registered rather than checking and adding when the properties are initialised for each element. The difficulty was refactoring linked attribute names to be calculated as soon as possible. This can be done, but requires a little more effort than this right here. I'd like to do that before merging this PR.